rivus codegen: Class fields default to private instead of inheriting class visibility

## Summary

When rivus generates TypeScript for class fields, fields without explicit visibility annotations default to `private` instead of inheriting visibility from the parent class. This differs from faber's behavior, where public classes produce public fields by default. The result is that rivus-compiled classes have private fields that cannot be accessed from factory functions or external code.

## Symptom

The artifex build (`bun run build:artifex`) fails type-checking with errors like:

```
error TS2341: Property 'species' is private and only accessible within class 'Symbolum'.
error TS2341: Property 'valor' is private and only accessible within class 'Symbolum'.
error TS2352: Conversion of type '{ species: ..., valor: ... }' to type 'Symbolum' may be a mistake...
  Property 'species' is private in type 'Symbolum' but not in type '{ ... }'.
```

**Comparison of same source compiled by faber vs rivus:**

Source (`fons/rivus/ast/lexema.fab`):
```fab
@ publicum
genus Symbolum {
    SymbolumGenus species
    textus valor
    ...
}
```

Faber output (`opus/rivus/fons/ts/ast/lexema.ts`):
```typescript
export class Symbolum {
  species: SymbolumGenus;    // No modifier = public (TS default)
  valor: string;
  ...
}
```

Rivus output (`opus/artifex/fons/ts/ast/lexema.ts`):
```typescript
export class Symbolum {
  private species: SymbolumGenus;    // Wrong!
  private valor: string;
  ...
}
```

## Root Cause

`fons/rivus/codegen/ts/sententia/genus.fab` in `genCampus()` (lines 164-171):

```fab
si campus.visibilitas == Visibilitas.Publica {
    result = scriptum("§public ", result)
} sin campus.visibilitas == Visibilitas.Protecta {
    result = scriptum("§protected ", result)
} secus {
    result = scriptum("§private ", result)   # Default fallback
}
```

The rivus codegen does not receive or use the parent class's visibility. It only checks the field's own `visibilitas`, which defaults to `Privata` when no annotation is present.

**Faber's approach** (`fons/faber/codegen/ts/statements/genus.ts`, lines 165-175):
1. Pass `classVisibility` to `genFieldDeclaration()`
2. Check if field has explicit visibility annotation
3. Use `effectiveVisibility = hasFieldVisibility ? fieldVisibility : classVisibility`
4. Emit modifier only for private/protected (public is TS default)

## Causal Chain

1. Source declares `@ publicum genus Symbolum { textus nomen ... }`
2. Parser creates `GenusDeclaratio` with class-level `visibilitas = Publica`
3. Parser creates `CampusDeclaratio` for `nomen` with `visibilitas = Privata` (default)
4. Rivus codegen calls `genGenus()` which iterates fields
5. `genCampus()` only sees `campus.visibilitas = Privata`, emits `private`
6. Generated TS has private fields despite public class
7. Factory functions using object literals can't construct instances

## Affected Components

- `fons/rivus/codegen/ts/sententia/genus.fab` - `genCampus()` missing inheritance
- `fons/rivus/parser/sententia/declara.fab` - sets field default (not wrong, but related)
- All rivus-compiled classes with implicit field visibility

## Fix Considerations

### Approach A: Pass class visibility to genCampus

**Changes Required:**
- Modify `genCampus` signature to accept parent visibility
- Update `genGenus` to pass class visibility when iterating fields
- Logic: if field has no explicit annotation, use class visibility

```fab
functio genCampus(CampusDeclaratio campus, Visibilitas? classVisibilitas, TsGenerator g) -> textus {
    varia effectiveVis = campus.visibilitasExplicita sic campus.visibilitas secus (classVisibilitas vel Visibilitas.Privata)
    # ... use effectiveVis
}
```

**Risk Areas:**
- Requires tracking whether field visibility was explicit (AST change)
- Current AST doesn't distinguish explicit from default visibility

### Approach B: Change field default from Privata to inherit

**Changes Required:**
- Make `CampusDeclaratio.visibilitas` nullable
- Interpret `nihil` as "inherit from class"
- Update parser to only set visibility when explicit

**Risk Areas:**
- Larger change touching parser and codegen
- Need to update all code checking `campus.visibilitas`

### Approach C: Match faber - check annotations in codegen

**Changes Required:**
- Store field annotations in AST (may already exist)
- In `genCampus`, check if any visibility annotation present
- If not, inherit from class visibility passed as parameter

**Risk Areas:**
- Requires annotation access in codegen

## Open Questions

- Does `CampusDeclaratio` store its annotations? If so, codegen can check directly.
- Should this fix also apply to methods? (Currently methods default to public correctly)
- Is there precedent in the language spec for visibility inheritance?

--- a/fons/rivus/codegen/ts/sententia/genus.fab
+++ b/fons/rivus/codegen/ts/sententia/genus.fab
@@ -26,6 +26,12 @@ ex "../typus" importa genTypus
 functio genGenus(textus nomen, ignotum generaParametra, ignotum extendit, ignotum implet, bivalens abstractum, lista<CampusDeclaratio> campi, ignotum structor, lista<Sententia> methodi, ignotum visibilitas, TsGenerator g) -> textus {
     varia result = g.ind()
 
+    # Determine class visibility for field inheritance
+    varia classVisibility = Visibilitas.Privata
+    si nonnihil visibilitas {
+        classVisibility = visibilitas qua Visibilitas
+    }
+
     # Module-level: export when public
     si nonnihil visibilitas {
         fixum vis = visibilitas qua Visibilitas
@@ -67,7 +73,7 @@ functio genGenus(textus nomen, ignotum generaParametra, ignotum extendit, ignotu
 
     # Fields
     ex campi pro campus {
-        result = scriptum("§§\n", result, genCampus(campus, g))
+        result = scriptum("§§\n", result, genCampus(campus, classVisibility, g))
     }
 
     # Split creo method from other methods
@@ -158,17 +164,28 @@ functio genCreoMethod(Sententia creo, TsGenerator g) -> textus {
 }
 
 @ publica
-functio genCampus(CampusDeclaratio campus, TsGenerator g) -> textus {
+functio genCampus(CampusDeclaratio campus, Visibilitas classVisibility, TsGenerator g) -> textus {
     varia result = g.ind()
 
-    # Visibility
-    si campus.visibilitas == Visibilitas.Publica {
-        result = scriptum("§public ", result)
-    } sin campus.visibilitas == Visibilitas.Protecta {
-        result = scriptum("§protected ", result)
-    } secus {
-        result = scriptum("§private ", result)
+    # Determine effective visibility
+    # HEURISTIC: For public classes, only emit explicit private/protected modifiers
+    # This matches TypeScript's default where public is implicit
+    varia effectiveVisibility = campus.visibilitas
+
+    # If class is public and field would be private (default), make it public instead
+    # This implements field visibility inheritance from parent class
+    si classVisibility == Visibilitas.Publica et campus.visibilitas == Visibilitas.Privata {
+        # Fields without explicit visibility in public classes should be public
+        effectiveVisibility = Visibilitas.Publica
     }
+
+    # Only emit modifier if private or protected (public is default in TS)
+    si effectiveVisibility == Visibilitas.Privata {
+        result = scriptum("§private ", result)
+    } sin effectiveVisibility == Visibilitas.Protecta {
+        result = scriptum("§protected ", result)
+    }
+    # For public visibility, emit no modifier (public is implicit in TypeScript)
 
     si campus.staticum {
         result = scriptum("§static ", result)

